tests/serverpod_test_server/test_e2e/database_roundtrip_type_test.dart

```diff
@@ -0,0 +1,194 @@
+import 'package:serverpod_test_client/serverpod_test_client.dart';
+import 'package:serverpod_test_server/test_util/config.dart';
+import 'package:test/test.dart';
+
+void main() {
+  var client = Client(serverUrl);
+
+  group('Given the "Type" database-roundtrip/echo endpoint', () {
+    test(
+        'When sending a new to be stored, then the count of stored objects is increased by one',
+        () async {
+      var types = Types();
+
+      var count = await client.basicDatabase.countTypesRows();
+      expect(count, isNotNull);
+
+      types = await client.basicDatabase.insertTypes(types);
+      expect(types.id, isNotNull);
+
+      var newCount = await client.basicDatabase.countTypesRows();
+      expect(newCount, isNotNull);
+      expect(newCount, equals(count! + 1));
+    });
+
+    // TODO: Support ByteData in database store
```

Comment:
Suggestion: fix this todo.

Reply:
I think this should be fully supported! But looks scary if we do not have any tests in place! I prefer to have these "TODOs" as issues in the repo, over comments like this, as todos are easily forgotten! Feel free to create issues and delete todos if you run into them!

Reply:
Hmm, maybe the reason this wasn't done, was because it doesn't work yet.

At least my basic test cases result in a surprising behavior

![CleanShot 2025-02-03 at 12 18 24@2x](https://github.com/user-attachments/assets/a121c50f-378e-4107-8aa1-6df696a25978)


![CleanShot 2025-02-03 at 12 17 13@2x](https://github.com/user-attachments/assets/73cd93c7-6094-4070-b48d-7b78d96cf1de)


Maybe I will have to carve this out into a follow-up, as I'll need to dig deeper to see what's going on here.

Reply:
Interestingly the E2E test going through the database and back out works, while the integration test going just into the database and back out does not 🤯

Maybe something in the extra to/from JSON that might be happening over the protocol corrects something 🤔 (Because the long data above looks like it would be looking at some encoded/wrapped info.)

Reply:
Ah, turns out that the `ByteData` in the "read from DB" case is pointing to a shared buffer, and my test so far was wrongly checking the whole buffer instead of just reading the assigned part (accessing which is trickier than I think such a standard operation should be, as this would also be what every consumer has to do).

With this updated assertion the DB integration tests now also passes:

```dart
    expect(
      Uint8List.view(
        retrieved!.aByteData!.buffer,
        retrieved.aByteData!.offsetInBytes,
        retrieved.aByteData!.lengthInBytes,
      ).toList(),
      [1, 2, 3],
    );
```

But I wonder if we couldn't provide a better "wrapper" on that field, such that such a mixup would not happen.

Reply:
Yeah, ByteData is easy to get wrong when it comes to the views. Maybe possible to make some abstraction but I have no good ideas for that 😅.